===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(p11_asn1_encode:der)) at 10 (line 185, column 6) }
===== Actual patch in diff format =====
@@ -183,6 +183,7 @@ p11_asn1_encode (node_asn *asn,
 	}
 
 	if (ret != ASN1_SUCCESS) {
+		free(der);
 		p11_debug_precond ("failed to encode: %s\n", message);
 		return NULL;
 	}
